module/zfs/zil.c

```diff
@@ -3469,6 +3612,95 @@ zil_commit_itx_assign(zilog_t *zilog, zil_commit_waiter_t *zcw)
 	dmu_tx_commit(tx);
 }
 
+/*
+ * Crash the ZIL. This is something like suspending, but abandons the ZIL
+ * without further IO until the wanted txg completes. No effort is made to
+ * close the on-disk chain or do any other on-disk work, as the pool may
+ * have suspended. zil_sync() will handle cleanup as normal and restart the
+ * ZIL once enough txgs have passed.
+ */
+static void
+zil_crash(zilog_t *zilog)
+{
+	mutex_enter(&zilog->zl_lock);
+
+	uint64_t txg = spa_syncing_txg(zilog->zl_spa);
+	uint64_t restart_txg = spa_syncing_txg(zilog->zl_spa) + TXG_SIZE;
```

Comment:
I don't think `TXG_SIZE` makes sense here. It should probably be `TXG_CONCURRENT_STATES` or `TXG_DEFER_SIZE`, haven't thought too deep.  I.e. there are only 3 concurrent TXG at all, not 4.  4 is just convenience for array sizes.